rust/xaynet-server/src/rest.rs

```diff
@@ -22,10 +23,17 @@ use xaynet_core::{crypto::ByteObject, ParticipantPublicKey};
 /// data and POST requests containing PET messages.
 ///
 /// * `api_settings`: address of the server and optional certificate and key for TLS server
-///   authentication.
+///   authentication as well as trusted anchors for TLS client authentication.
 /// * `fetcher`: fetcher for responding to data requests.
 /// * `pet_message_handler`: handler for responding to PET messages.
-pub async fn serve<F>(api_settings: ApiSettings, fetcher: F, pet_message_handler: PetMessageHandler)
+///
+/// # Errors
+/// Fails if invalid TLS settings are provided when the `tls` feature is enabled.
+pub async fn serve<F>(
+    api_settings: ApiSettings,
+    fetcher: F,
+    pet_message_handler: PetMessageHandler,
+) -> Option<()>
```

Comment:
was going to say, wouldn't a `bool` return type do just as well? but perhaps `Option` is more convenient with the `?` syntax ...

Reply:
that's indeed just for `?` error handling from within the `run_...` methods. later on we should probably replace the `Option` with a `Result` and add some proper error handling to the `serve` method.

Reply:
i introduced an error type right away during the rebase.